#!/usr/bin/env python
# http://wiki.ros.org/dynamic_reconfigure/Tutorials/HowToWriteYourFirstCfgFile
PACKAGE = "people_detection_vision"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("resize_max_width",    int_t,    0, "Image max width (px)",      320,  160, 800)
gen.add("resize_max_height",   int_t,    0, "Image max height (px)",     320,  160, 800)
gen.add("scale_factor",        double_t, 0, "Scale factor",   1.2,  1.01, 1.5)
gen.add("min_neighbors",       int_t,    0, "Min neighbours",  1,   0, 3)
gen.add("min_width",           int_t,    0, "Min width (px)",       10,   0, 50)

gen.add("max_sample_depth",        double_t, 0, "Reprojected face max width (m)",   0.2,  0, 1)
gen.add("max_sample_width",        double_t, 0, "Reprojected face max height (m)",  0.15, 0, 1)

exit(gen.generate(PACKAGE, "people_detection_vision", "Viola"))
